src/deploy.ts

```diff
@@ -86,51 +102,121 @@ export class HypERC20Deployer extends GasRouterDeployer<
     );
   }
 
+  protected async deployCollateral(
+    chain: ChainName,
+    config: CollateralConfig & HypERC20Config,
+  ): Promise<HypERC20Collateral> {
+    this.logger(`Deploying collateral router on ${chain}...`);
+    const erc20 = new ERC20__factory()
+      .attach(config.token)
+      .connect(this.multiProvider.getProvider(chain));
+
+    const decimals = await erc20.decimals();
+    const name = await erc20.name();
+    const symbol = await erc20.symbol();
+    const totalSupply = 0; // synthetic tokens are minted on demand
+
+    this.tokenMetadata = { decimals, name, symbol, totalSupply };
```

Comment:
Personally do not like this. Don't feel strongly, but would slightly prefer something that automatically extends a config that the deployer can rely on or fetch from the collateral token during the synthetic token deploy branch. I know it's duplicative, but setting instance variables is brittle imo. How do you ensure that the collateral gets deployed first?

Reply:
Not sure I agree using a config is the way to go. Related, see Asa's complaint here about too much config duplication: https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/1863#issuecomment-1482107196

Should this be it's own method (e.g. `fetchTokenMetadata`) that any deployer method calls if tokenMetadata is not yet assigned?